src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -340,14 +353,31 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
 
                     using (FileStream fs = File.Create(codePath))
                     {
-                        MethodInfo method = typeof(System.Xml.Serialization.XmlSerializer).GetMethod("GenerateSerializer", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+                        MethodInfo method;
+                        if (defaultNamespace == null)
+                        {
+                            method = typeof(System.Xml.Serialization.XmlSerializer).GetMethod("GenerateSerializer", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+                        }
+                        else
+                        {
+                            Type tempAssemblyType = typeof(System.Xml.Serialization.XmlSerializer).Assembly.GetType("System.Xml.Serialization.TempAssembly");
```

Comment:
My first reaction was to think that this would be cleaner if the PR added an overload for 'GenerateSerializer' that accepted a namespace, since it ends up calling this next level method anyway. But I suspect that's old in-box framework thinking and is really a can of worms that best left alone since we may have mismatched assembly versions in this agile new world of .Net.

I might suggest adding BindingFlags.Public to cover your bases like the original code, just in case something changes.

Reply:
While I am willing to create a PR that adds an overload for 'GenerateSerializer' that accepts a namespace - it will not help in the immediate future (i.e. in this PR) - since it will only be available in v6.0 of the runtime.

GetType behaves differently than GetMethod, there is no need here for binding flags (in fact - no such overload exists).